app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailScroller.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.compose.plantdetail
+
+import androidx.animation.FloatPropKey
+import androidx.animation.Spring
+import androidx.animation.transitionDefinition
+import androidx.compose.Composable
+import androidx.ui.foundation.ScrollerPosition
+import androidx.ui.unit.Density
+import androidx.ui.unit.Dp
+
+private const val HeaderTransitionOffset = 150f
```

Comment:
I still think this should be in `Dp`s

Reply:
As per our previous conversation, having it `Dp`s makes it very tricky as this is used in `PlantDetailsScroller` where I don't have the density. Also, the rest of the equation to calculate the transition is in `Float`. 

Any suggestions to improve this?

Reply:
Since this is describing a height on the screen, this shouldn't be pixels. Perhaps pass the density or the transformed height to PlantDetailsScroller?

Reply:
Also I'm not finding the usage in this PR, is there a file missing? :)

Reply:
The usage is in this same file (line 40). While I agree with your thoughts, using `Dp`s here would be really messy and would hurt readability massively.

If we had this in `Dp`s, you'd need to pass it in and wrap the if-else block inside `with(density)`. What's more, is that the caller also needs to wrap the call to `PlantDetailsScroller` into a `with(DensityAmbient.current` to get this to work. 

I think it's too much for little benefit, IMO.

Reply:
I absolutely think we need to do this else you'll get different triggers depending upon device density! It should just require a simple conversion before usage `val offset = with(DensityAmbient.current) { HeaderTransitionOffset.toPx() }`

Reply:
Done! Sorry for the back and forth, added density to the class params and it's not _that_ ugly. Thanks!